File: app/src/main/java/com/testapp/testapp/presenter/PhotosPresenter.java
package com.testapp.testapp.presenter;

import android.support.annotation.NonNull;

import com.testapp.testapp.model.entity.Photo;
import com.testapp.testapp.model.entity.response.Response;
import com.testapp.testapp.model.entity.response.ResponsePhotos;
import com.testapp.testapp.model.rest.RestApiManager;
import com.testapp.testapp.presenter.utils.RequestParametersHolder;
import com.testapp.testapp.view.CustomListView;

import retrofit2.Call;
import retrofit2.Callback;

/**
 * Created by troll on 07.11.2017.
 */

public class PhotosPresenter implements Presenter, Callback<Response<ResponsePhotos>> {

    private CustomListView<Photo> viewForList;
    private String venueId;

    public PhotosPresenter(CustomListView<Photo> viewForList, String venueId) {
        this.viewForList = viewForList;
        this.venueId = venueId;
    }

    @Override
    public void getResponse() {
        Call<Response<ResponsePhotos>> responseCall = new RestApiManager()
                .getVenueApi()
                .venuePhotos(venueId, RequestParametersHolder.getInstance().getVenuePhotosParams());
        responseCall.enqueue(this);
    }

    @Override
    public void onResponse(@NonNull Call<Response<ResponsePhotos>> call, @NonNull retrofit2.Response<Response<ResponsePhotos>> response) {
        Response<ResponsePhotos> commonResponse = response.body();
        if (commonResponse != null){
            ResponsePhotos responsePhotos = commonResponse.getResponse();
            viewForList.onSuccessResponse(responsePhotos.getPhotoWrapper().getItems());
        }
    }

    @Override
    public void onFailure(@NonNull Call<Response<ResponsePhotos>> call, @NonNull Throwable t) {
        viewForList.onFailureRequest();
    }
}

File: app/src/main/java/com/testapp/testapp/kotlin/presenter/PhotosPresenter.kt
package com.testapp.testapp.kotlin.presenter

import com.testapp.testapp.kotlin.model.entity.Photo
import com.testapp.testapp.kotlin.model.entity.response.Response
import com.testapp.testapp.kotlin.model.entity.response.ResponsePhotos
import com.testapp.testapp.kotlin.model.rest.RestApiManager
import com.testapp.testapp.kotlin.presenter.utils.RequestParametersHolder
import com.testapp.testapp.kotlin.view.CustomListView
import retrofit2.Call
import retrofit2.Callback

class PhotosPresenter(val viewForList: CustomListView<Photo>, val venueId: String) : Presenter, Callback<Response<ResponsePhotos>> {

    override fun getResponse() {
        val restApiManager = RestApiManager.instance
        val responseCall = restApiManager.getVenueApi()
                ?.venuePhotos(venueId, RequestParametersHolder.instance.getVenuePhotosParams())
        responseCall?.enqueue(this)
    }

    override fun onFailure(call: Call<Response<ResponsePhotos>>?, t: Throwable?) {
        viewForList.onFailureRequest()
    }

    override fun onResponse(call: Call<Response<ResponsePhotos>>?, response: retrofit2.Response<Response<ResponsePhotos>>?) {
        val commonResponse = response?.body()
        val responsePhotos = commonResponse?.response
        viewForList.onSuccessResponse(responsePhotos?.photoWrapper!!.items)
    }
}